src/components/Select/index.tsx

```diff
@@ -0,0 +1,272 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable react/jsx-props-no-spreading,
+@typescript-eslint/no-explicit-any,
+react/destructuring-assignment */
+import { FC, useCallback, useMemo } from 'react';
+import RSelect, {
+  components as component,
+  ControlProps,
+  MenuProps,
+  OptionProps,
+  PlaceholderProps,
+  IndicatorsContainerProps,
+} from 'react-select';
+import cx from 'classnames';
+import { OptionType, SelectColor, SelectProps, SelectSize } from 'types';
+
+import { Text } from '..';
+
+import styles from './styles.module.scss';
+import { arrow } from 'assets/img';
+
+const ROOT = document.querySelector('body');
+
+type Props = {
+  size?: SelectSize;
+  color?: SelectColor;
+} & SelectProps;
+
+const Select: FC<Props> = ({
```

Comment:
Why do you need all these huge amount of properties? Do you really would use them?